hooks/000-provide-uids-gids.chroot

```diff
@@ -12,67 +12,67 @@
 
 echo "Ensure passwd file is in a sane state and did not change"
 diff -u /etc/passwd - <<EOF
-root:x:0:0:root:/root:/bin/bash
-daemon:x:1:1:daemon:/usr/sbin:/usr/sbin/nologin
-bin:x:2:2:bin:/bin:/usr/sbin/nologin
-sys:x:3:3:sys:/dev:/usr/sbin/nologin
-sync:x:4:65534:sync:/bin:/bin/sync
-games:x:5:60:games:/usr/games:/usr/sbin/nologin
-man:x:6:12:man:/var/cache/man:/usr/sbin/nologin
-lp:x:7:7:lp:/var/spool/lpd:/usr/sbin/nologin
-mail:x:8:8:mail:/var/mail:/usr/sbin/nologin
-news:x:9:9:news:/var/spool/news:/usr/sbin/nologin
-uucp:x:10:10:uucp:/var/spool/uucp:/usr/sbin/nologin
-proxy:x:13:13:proxy:/bin:/usr/sbin/nologin
-www-data:x:33:33:www-data:/var/www:/usr/sbin/nologin
-backup:x:34:34:backup:/var/backups:/usr/sbin/nologin
-list:x:38:38:Mailing List Manager:/var/list:/usr/sbin/nologin
-irc:x:39:39:ircd:/run/ircd:/usr/sbin/nologin
-_apt:x:42:65534::/nonexistent:/usr/sbin/nologin
-nobody:x:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
+root:*:0:0:root:/root:/bin/bash
```

Comment:
I am not sure this is normal to use `*` here, unless you removed them from `/etc/shadow`. But I do not think that would be correct either.

Reply:
The reason I changed this is because the `/etc/passwd` that now comes from the deb package through chisel looks like this (it's merely a copy of `/usr/share/base-passwd/group.master`). Which means this diff check fails here. I looked up the `*` modifier and it means (from manpages)

```
If the password field contains some string that is not a valid result
of crypt(3), for instance ! or *, the user will not be able to use a
unix password to log in (but the user may log in the system by other
means).
```

Reply:
Right. It is not where we want to say that. We want `*` that in `/etc/shadow`. We need `x` in `/etc/passwd` to say we are using `/etc/shadow`. Unprivileged users do not have to know if a password has been set for the users. `*` gives too much information.

Reply:
I'll mark this as resolved as we write our own /etc/* files